libc/signal.c

```diff
@@ -0,0 +1,199 @@
+/*
+ * Phoenix-RTOS
+ *
+ * POSIX.1-2017 standard library functions tests
+ * HEADER:
+ *    - signal.h
+ *
+ * TESTED:
+ *    - psignal()
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Mateusz Bloch
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+#include <signal.h>
+#include <limits.h>
+#include <unistd.h>
+#include <unity_fixture.h>
+
+/* Typical error message does not exceed ~60 characters, that's why we expect a maximum value a little bit bigger */
+#define MAX_LEN_STRING 100
+#define STDERR_PATH    "stderr.log"
+#define SIMPLE_MSG     "Simple Message"
+
+static FILE *err_file;
+static int stderr_fd;
+
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wunused-const-variable"
+static const int signal_codes[] = { SIGABRT, SIGALRM, SIGBUS, SIGCHLD, SIGCONT, SIGFPE, SIGHUP, SIGILL, SIGINT,
+	SIGKILL, SIGPIPE, SIGQUIT, SIGSEGV, SIGSTOP, SIGTERM, SIGTSTP, SIGTTIN, SIGTTOU, SIGUSR1, SIGUSR2, SIGPROF,
+	SIGSYS, SIGTRAP, SIGURG, SIGVTALRM, SIGXCPU, SIGXFSZ };
+
+static const unsigned int signal_codes_len = sizeof(signal_codes) / sizeof(signal_codes[0]);
+#pragma GCC diagnostic pop
+
+TEST_GROUP(signal_psignal);
+
+TEST_SETUP(signal_psignal)
+{
+	err_file = fopen(STDERR_PATH, "a+");
+	stderr_fd = dup(STDERR_FILENO);
+}
+
+
+TEST_TEAR_DOWN(signal_psignal)
+{
+	TEST_ASSERT_NOT_NULL(freopen("/dev/stderr", "w", stderr));
+	close(STDERR_FILENO);
```

Comment:
`stderr` should be unbuffered but still calling `fflush(stderr)` beforehand is a good idea.

You may want to write some comment to future yourself here or in TEST_SETUP regarding what you're doing (or why)  ;)